code/game/objects/items/devices/illegalradio.dm

```diff
@@ -194,74 +233,65 @@
 	dat = jointext(dat,"") 
 	return dat	
 
+
+	
+	
 /obj/item/device/illegalradio/proc/dispense_change()
 	if(money_stored > 0)
 		dispense_cash(money_stored,get_turf(src))
 		money_stored = 0
 	interact(usr)
 
 /obj/item/device/illegalradio/afterattack(atom/A as mob|obj, mob/user as mob)
-	if(istype(A, /obj/item/weapon/spacecash))
+	if(istype(A, /obj/item/weapon/spacecash) && A.Adjacent(user))
 		var/obj/item/weapon/spacecash/cash = A
 		money_stored += cash.get_total()
 		qdel(cash)
 		visible_message("<span class='info'>[usr] inserts a credit chip into [src].</span>")
 		interact(usr)
-	else
-		if(istype(A, /obj))
-			attempt_sell(A,usr)
+	else if((istype(A, /obj) || istype(A, /mob)) && A.Adjacent(user) && !scanning)
+		scanning = 1
+		attempt_sell(A,usr)
+		scanning = 0
 			
 /obj/item/device/illegalradio/proc/attempt_sell(var/obj/input, mob/user)	//If statements galore
-	visible_message("The uplink beeps: <span class='warning'>Scanning item...</span>")
+	visible_message("The [name] beeps: <span class='warning'>Scanning item...</span>")
```

Comment:
`[src]`?, also applies to all line below

Reply:
name is actually ok here, when you use [name] it does not generate a "the" if the name is improper

it's more popular to do \The [src] in general because a \the macro will delete the preceding the if it would be followed by a the - but this should appear fine

Reply:
*shrug*